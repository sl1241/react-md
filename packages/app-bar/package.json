{
  "name": "@react-md/app-bar",
  "version": "5.1.3",
  "description": "This small package implments the AppBar spec in material design.",
  "main": "./lib/index.js",
  "module": "./es/index.js",
  "types": "./types/index.d.ts",
  "sideEffects": [
    "dist/**/*"
  ],
  "author": "Mikkel Laursen <mlaursen03@gmail.com>",
  "repository": {
    "type": "git",
    "url": "https://github.com/mlaursen/react-md.git",
    "directory": "packages/app-bar"
  },
  "bugs": {
    "url": "https://github.com/mlaursen/react-md/issues"
  },
  "homepage": "https://react-md.dev/packages/app-bar/demos",
  "keywords": [
    "react-md",
    "material design",
    "react",
    "app bar",
    "appbar",
    "toolbar",
    "component"
  ],
  "license": "MIT",
  "dependencies": {
    "@react-md/button": "workspace:../button",
    "@react-md/elevation": "workspace:../elevation",
    "@react-md/icon": "workspace:../icon",
    "@react-md/theme": "workspace:../theme",
    "@react-md/typography": "workspace:../typography",
    "@react-md/utils": "workspace:../utils",
    "classnames": "^2.3.2"
  },
  "devDependencies": {
    "@types/react": "^18.0.29",
    "csstype": "^3.1.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "peerDependencies": {
    "react": ">= 16.14",
    "react-dom": ">= 16.14"
  },
  "peerDependenciesMeta": {
    "react": {
      "optional": false
    },
    "react-dom": {
      "optional": false
    }
  },
  "publishConfig": {
    "access": "public"
  }
}
