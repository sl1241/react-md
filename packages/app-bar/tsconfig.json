{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist/types"
  },
  "exclude": [
    "vite.config.ts",
    "vite.config.umd.ts",
    "./src ori",
    "**/dist/**",
    "**/lib/**",
    "**/types/**",
    "**/__tests__/**"
  ],
  "references": [
    {
      "path": "../button"
    },
    {
      "path": "../elevation"
    },
    {
      "path": "../icon"
    },
    {
      "path": "../theme"
    },
    {
      "path": "../typography"
    },
    {
      "path": "../react"
    },
    {
      "path": "../utils"
    }
  ]
}
