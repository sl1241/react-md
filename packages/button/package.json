{
  "name": "@react-md/button",
  "version": "5.1.3",
  "description": "This package is used to create buttons with the material design spec. There is built-in support to be able to render accessible clickable divs that look like buttons, or even apply button styles to any element.",
  "main": "./lib/index.js",
  "module": "./es/index.js",
  "types": "./types/index.d.ts",
  "sideEffects": [
    "dist/**/*"
  ],
  "author": "Mikkel Laursen <mlaursen03@gmail.com>",
  "repository": {
    "type": "git",
    "url": "https://github.com/mlaursen/react-md.git",
    "directory": "packages/button"
  },
  "bugs": {
    "url": "https://github.com/mlaursen/react-md/issues"
  },
  "homepage": "https://react-md.dev/packages/button/demos",
  "keywords": [
    "accessibility",
    "a11y",
    "react-md",
    "material design",
    "react",
    "button",
    "component",
    "components"
  ],
  "license": "MIT",
  "dependencies": {
    "@react-md/elevation": "workspace:../elevation",
    "@react-md/icon": "workspace:../icon",
    "@react-md/states": "workspace:../states",
    "@react-md/theme": "workspace:../theme",
    "@react-md/transition": "workspace:../transition",
    "@react-md/typography": "workspace:../typography",
    "@react-md/utils": "workspace:../utils",
    "classnames": "^2.3.2"
  },
  "devDependencies": {
    "@types/react": "^18.0.29",
    "csstype": "^3.1.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "peerDependencies": {
    "react": ">= 16.14",
    "react-dom": ">= 16.14"
  },
  "peerDependenciesMeta": {
    "react": {
      "optional": false
    },
    "react-dom": {
      "optional": false
    }
  },
  "publishConfig": {
    "access": "public"
  }
}
