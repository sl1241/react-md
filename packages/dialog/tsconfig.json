{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist/types"
    //"composite": true,
  },
  "include": ["./src"],
  "exclude": [
    "vite.config.ts",
    "vite.config.umd.ts",
    "./src ori",
    "**/dist/**",
    "**/lib/**",
    "**/types/**",
    "**/__tests__/**"
  ],
  "references": [
    {
      "path": "../app-bar"
    },
    {
      "path": "../elevation"
    },
    {
      "path": "../overlay"
    },
    {
      "path": "../portal"
    },
    {
      "path": "../theme"
    },
    {
      "path": "../transition"
    },
    {
      "path": "../typography"
    },
    {
      "path": "../react"
    },
    {
      "path": "../utils"
    }
  ]
}
