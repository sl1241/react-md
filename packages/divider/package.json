{
  "name": "@react-md/divider",
  "version": "5.1.3",
  "description": "This package is used to create horizontal or vertical dividers in your application.",
  "main": "./lib/index.js",
  "module": "./es/index.js",
  "types": "./types/index.d.ts",
  "sideEffects": [
    "dist/**/*"
  ],
  "author": "Mikkel Laursen <mlaursen03@gmail.com>",
  "repository": {
    "type": "git",
    "url": "https://github.com/mlaursen/react-md.git",
    "directory": "packages/divider"
  },
  "bugs": {
    "url": "https://github.com/mlaursen/react-md/issues"
  },
  "homepage": "https://react-md.dev/packages/divider/demos",
  "keywords": [
    "react-md",
    "material design",
    "react",
    "divider",
    "component",
    "components"
  ],
  "license": "MIT",
  "dependencies": {
    "@react-md/theme": "workspace:../theme",
    "@react-md/utils": "workspace:../utils",
    "classnames": "^2.3.2"
  },
  "devDependencies": {
    "@types/react": "^18.0.29",
    "csstype": "^3.1.1",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "peerDependencies": {
    "react": ">= 16.14",
    "react-dom": ">= 16.14"
  },
  "peerDependenciesMeta": {
    "react": {
      "optional": false
    },
    "react-dom": {
      "optional": false
    }
  },
  "publishConfig": {
    "access": "public"
  }
}
