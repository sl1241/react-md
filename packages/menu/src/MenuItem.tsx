

import { ListItem } from "@react-md/list"
import { useKeyboardFocusableElement } from "@react-md/utils"

import type { MenuItemProps } from "./types"

/**
 * This component is used as an "action" within a `Menu`/`DropdownMenu` that
 * implements some keyboard focus behavior. This component should generally have
 * an `onClick` event handler.
 *
 * @remarks \@since 5.0.0
 */
export const MenuItem = ({ className, children, role = "menuitem", tabIndex = -1, ref: nodeRef, ...props }: MenuItemProps) => {
  const ref = useKeyboardFocusableElement(nodeRef as any)
  return (
    <ListItem
      {...props}
      ref={ref}
      //@ts-ignore
      role={role}
      tabIndex={tabIndex}
      className={["rmd-menu-item", className]}
    >
      {children}
    </ListItem>
  )
}

