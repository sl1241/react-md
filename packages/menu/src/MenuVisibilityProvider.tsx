
import { createContext, useContext, useMemo } from 'voby'
import type { HoverModeHookReturnValue } from "@react-md/utils"

/** @remarks \@since 5.0.0 */
export type MenuVisibilityContext = Pick<HoverModeHookReturnValue, "visible">

/**
 * @internal
 * @remarks \@since 5.0.0
 */
const context = createContext<MenuVisibilityContext>({
  //@ts-ignore
  visible: false,
  // setVisible() {
  //   throw new Error('"MenuVisibilityProvider" must be a parent component')
  // },
})
//@ts-ignore
context.displayName = "MenuVisibility"

/**
 * @internal
 * @remarks \@since 5.0.0
 */
const { Provider } = context

/**
 * This hook allows you control the visibility of a parent menu. The main
 * use-case for this hook is adding a custom sheet header/footer.
 *
 * @example
 * Simple Example
 * ```tsx
 * function SheetFooter(): Element {
 *   const { setVisible } = useMenuVisibility();
 *
 *   return (
 *     <DialogFooter>
 *       <Button onClick={() => setVisible(false)}>Cancel</Button>
 *     </DialogFooter>
 *   );
 * }
 * ```
 *
 * @returns the {@link MenuVisibilityContext}
 * @remarks \@since 5.0.0
 */
export function useMenuVisibility(): Readonly<MenuVisibilityContext> {
  return useContext(context)
}

/**
 * @internal
 * @remarks \@since 5.0.0
 */
export interface MenuVisibilityProviderProps extends MenuVisibilityContext {
  children: Children
}

/**
 * @internal
 * @remarks \@since 5.0.0
 */
export function MenuVisibilityProvider({ visible, children, }: MenuVisibilityProviderProps): Element {
  const value = useMemo<MenuVisibilityContext>(() => ({ visible, }))

  return <Provider value={value}>{children}</Provider>
}
