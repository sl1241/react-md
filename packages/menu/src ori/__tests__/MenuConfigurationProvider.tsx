import { AppBar, AppBarAction, AppBarTitle } from "@react-md/app-bar";
import { Button } from "@react-md/button";
import { DialogFooter } from "@react-md/dialog";
import { Divider } from "@react-md/divider";
import { Configuration } from "@react-md/layout";
import { CloseSVGIcon } from "@react-md/material-icons";
import {
  DEFAULT_DESKTOP_LARGE_MIN_WIDTH,
  DEFAULT_DESKTOP_MIN_WIDTH,
  DEFAULT_PHONE_MAX_WIDTH,
  DEFAULT_TABLET_MAX_WIDTH,
} from "@react-md/utils";
import type { RenderResult } from "@testing-library/react";
import {
  fireEvent,
  render as baseRender,
  waitFor,
} from "@testing-library/react";
import type { FC, ReactElement } from "react";

import { DropdownMenu } from "../DropdownMenu";
import { MenuConfigurationProvider } from "../MenuConfigurationProvider";
import { MenuItem } from "../MenuItem";
import { MenuItemLink } from "../MenuItemLink";
import { useMenuVisibility } from "../MenuVisibilityProvider";
import type { MenuConfiguration } from "../types";

const Wrapper: FC = ({ children }) => (
  <Configuration disableRipple>{children}</Configuration>
);

function render(ui: ReactElement): RenderResult {
  return baseRender(ui, { wrapper: Wrapper });
}

interface TestProps extends MenuConfiguration {
  horizontal?: boolean;
}

function Test(props: TestProps): ReactElement {
  return (
    <DropdownMenu {...props} id="main-menu" buttonChildren="Dropdown">
      <MenuItem>An Item</MenuItem>
      <DropdownMenu {...props} id="sub-menu" buttonChildren="Submenu">
        <MenuItem>Subitem 1</MenuItem>
        <MenuItem>Subitem 2</MenuItem>
        <MenuItem>Subitem 3</MenuItem>
      </DropdownMenu>
      <MenuItemLink href="#">Link</MenuItemLink>
      <MenuItem disabled>Disabled Item</MenuItem>
      <MenuItem>Delta</MenuItem>
    </DropdownMenu>
  );
}

function mockMedia(
  size: "phone" | "tablet" | "desktop"
): jest.SpyInstance<MediaQueryList, [query: string]> {
  let regexp: RegExp;
  switch (size) {
    case "phone":
      regexp = new RegExp(`max-width: ${DEFAULT_PHONE_MAX_WIDTH}`);
      break;
    case "tablet":
      regexp = new RegExp(`max-width: ${DEFAULT_TABLET_MAX_WIDTH}`);
      break;
    case "desktop":
      regexp = new RegExp(
        `min-width: (${DEFAULT_DESKTOP_MIN_WIDTH}|${DEFAULT_DESKTOP_LARGE_MIN_WIDTH})`
      );
      break;
  }

  return jest.spyOn(window, "matchMedia").mockImplementation((query) => ({
    media: query,
    matches: regexp.test(query),
    onchange: () => {},
    addListener: () => {},
    removeListener: () => {},
    addEventListener: () => {},
    removeEventListener: () => {},
    dispatchEvent: () => false,
  }));
}

describe("MenuConfigurationProvider", () => {
  describe('renderAsSheet="phone"', () => {
    let mock: ReturnType<typeof mockMedia>;
    beforeAll(() => {
      mock = mockMedia("phone");
    });

    afterAll(() => {
      mock?.mockRestore();
    });

    it("should allow menus to be rendered in a sheet on phones", async () => {
      const { getByRole } = render(
        <MenuConfigurationProvider renderAsSheet="phone">
          <Test />
        </MenuConfigurationProvider>
      );

      const dropdown = getByRole("button", { name: "Dropdown" });
      fireEvent.click(dropdown);
      expect(document.body).toMatchSnapshot();

      const sheet = getByRole("dialog", { name: "Dropdown" });
      const menu = getByRole("menu", { name: "Dropdown" });
      await waitFor(() => {
        expect(document.activeElement).toBe(menu);
      });

      fireEvent.click(getByRole("menuitem", { name: "Delta" }));
      await waitFor(() => {
        expect(menu).not.toBeInTheDocument();
        expect(sheet).not.toBeInTheDocument();
      });
      expect(document.body).toMatchSnapshot();
    });
  });

  it("should allow menus to be rendered in a sheet by enabling renderAsSheet", async () => {
    const { getByRole, rerender } = render(
      <MenuConfigurationProvider renderAsSheet>
        <Test />
      </MenuConfigurationProvider>
    );

    const dropdown = getByRole("button", { name: "Dropdown" });
    fireEvent.click(dropdown);

    expect(document.body).toMatchSnapshot();
    const sheet = getByRole("dialog", { name: "Dropdown" });
    const menu = getByRole("menu", { name: "Dropdown" });
    await waitFor(() => {
      expect(document.activeElement).toBe(menu);
    });

    fireEvent.click(getByRole("menuitem", { name: "Delta" }));
    await waitFor(() => {
      expect(menu).not.toBeInTheDocument();
      expect(sheet).not.toBeInTheDocument();
    });
    expect(document.body).toMatchSnapshot();

    rerender(
      <MenuConfigurationProvider renderAsSheet>
        <Test renderAsSheet={false} />
      </MenuConfigurationProvider>
    );
    fireEvent.click(dropdown);
    expect(document.body).toMatchSnapshot();
    expect(() => getByRole("dialog")).toThrow();
  });

  it("should allow for a custom header and footer for the sheet", async () => {
    function Header(): ReactElement {
      const { setVisible } = useMenuVisibility();
      return (
        <AppBar theme="clear">
          <AppBarTitle>Custom</AppBarTitle>
          <AppBarAction first last onClick={() => setVisible(false)}>
            <CloseSVGIcon />
          </AppBarAction>
        </AppBar>
      );
    }

    function Footer(): ReactElement {
      const { setVisible } = useMenuVisibility();
      return (
        <>
          <Divider />
          <DialogFooter>
            <Button onClick={() => setVisible(false)}>Cancel</Button>
          </DialogFooter>
        </>
      );
    }

    const { getByRole, rerender } = render(
      <MenuConfigurationProvider
        renderAsSheet
        sheetHeader={<Header />}
        sheetFooter={<Footer />}
      >
        <Test />
      </MenuConfigurationProvider>
    );

    const dropdown = getByRole("button", { name: "Dropdown" });
    fireEvent.click(dropdown);

    let sheet = getByRole("dialog", { name: "Dropdown" });
    let menu = getByRole("menu", { name: "Dropdown" });
    await waitFor(() => {
      expect(document.activeElement).toBe(menu);
    });
    expect(sheet).toMatchSnapshot();
    expect(() => getByRole("heading", { name: "Custom" })).not.toThrow();
    expect(() => getByRole("separator")).not.toThrow();

    // shouldn't close
    fireEvent.click(getByRole("heading", { name: "Custom" }));
    expect(sheet).toBeInTheDocument();
    expect(menu).toBeInTheDocument();

    const overlay = document.querySelector(".rmd-overlay");
    if (!overlay) {
      throw new Error();
    }
    fireEvent.click(overlay);
    await waitFor(() => {
      expect(sheet).not.toBeInTheDocument();
      expect(menu).not.toBeInTheDocument();
    });

    rerender(
      <MenuConfigurationProvider
        renderAsSheet
        sheetHeader={<Header />}
        sheetFooter={<Footer />}
      >
        <Test sheetHeader={null} sheetFooter={null} />
      </MenuConfigurationProvider>
    );
    fireEvent.click(dropdown);

    sheet = getByRole("dialog", { name: "Dropdown" });
    menu = getByRole("menu", { name: "Dropdown" });
    await waitFor(() => {
      expect(document.activeElement).toBe(menu);
    });
    expect(sheet).toMatchSnapshot();
    expect(() => getByRole("heading", { name: "Custom" })).toThrow();
    expect(() => getByRole("separator")).toThrow();
  });

  describe("useMenuVisibility", () => {
    const error = jest.spyOn(console, "error");
    beforeAll(() => {
      error.mockImplementation(() => {
        // do nothing
      });
    });

    afterAll(() => {
      error.mockRestore();
    });

    it("should throw an error if there isn't a parent MenuConfigurationProvider when the setVisible function is called", () => {
      function Test(): null {
        const { setVisible } = useMenuVisibility();
        setVisible(false);
        return null;
      }

      expect(() => render(<Test />)).toThrowError(
        '"MenuVisibilityProvider" must be a parent component'
      );
    });
  });
});
