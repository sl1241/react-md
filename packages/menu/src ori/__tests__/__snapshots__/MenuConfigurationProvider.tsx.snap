// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuConfigurationProvider renderAsSheet="phone" should allow menus to be rendered in a sheet on phones 1`] = `
<body
  class="rmd-utils--mouse"
  data-rmd-noscroll=""
  style="overflow: hidden;"
>
  <div>
    <button
      aria-expanded="true"
      aria-haspopup="menu"
      class="rmd-button rmd-button--text"
      id="main-menu"
      type="button"
    >
      Dropdown
      <i
        aria-hidden="true"
        class="rmd-icon rmd-icon--font material-icons rmd-icon-rotator rmd-icon-rotator--animate rmd-icon-rotator--rotated rmd-icon--after"
      >
        arrow_drop_down
      </i>
    </button>
  </div>
  <span
    class="rmd-overlay rmd-overlay--visible rmd-overlay--clickable rmd-dialog-overlay rmd-sheet-overlay rmd-overlay--active"
    id="main-menu-menu-sheet-overlay"
    tabindex="-1"
  />
  <div
    aria-labelledby="main-menu"
    aria-modal="true"
    class="rmd-dialog rmd-sheet rmd-sheet--vertical rmd-sheet--raised rmd-sheet--bottom rmd-sheet--touchable-height rmd-sheet--offscreen rmd-sheet--enter rmd-sheet--visible"
    id="main-menu-menu-sheet"
    role="dialog"
    tabindex="-1"
  >
    <div
      aria-labelledby="main-menu"
      class="rmd-menu"
      id="main-menu-menu"
      role="menu"
      tabindex="0"
    >
      <ul
        class="rmd-list"
        role="none"
      >
        <li
          class="rmd-list-item rmd-list-item--clickable rmd-menu-item"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="rmd-list-item__text"
          >
            An Item
          </span>
          <span
            class="rmd-ripple-container"
          />
        </li>
        <li
          aria-haspopup="menu"
          class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
          id="sub-menu"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="rmd-list-item__text"
          >
            Submenu
          </span>
          <i
            aria-hidden="true"
            class="rmd-icon rmd-icon--font material-icons rmd-icon-rotator rmd-icon-rotator--animate rmd-icon--after rmd-list-item__addon"
          >
            keyboard_arrow_right
          </i>
          <span
            class="rmd-ripple-container"
          />
        </li>
        <li
          role="none"
        >
          <a
            class="rmd-list-item rmd-list-item--clickable rmd-list-item--link rmd-menu-item"
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            Link
          </a>
        </li>
        <li
          aria-disabled="true"
          class="rmd-list-item rmd-list-item--clickable rmd-list-item--disabled rmd-list-item--disabled-color rmd-menu-item"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="rmd-list-item__text"
          >
            Disabled Item
          </span>
          <span
            class="rmd-ripple-container"
          />
        </li>
        <li
          class="rmd-list-item rmd-list-item--clickable rmd-menu-item"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="rmd-list-item__text"
          >
            Delta
          </span>
          <span
            class="rmd-ripple-container"
          />
        </li>
      </ul>
    </div>
  </div>
</body>
`;

exports[`MenuConfigurationProvider renderAsSheet="phone" should allow menus to be rendered in a sheet on phones 2`] = `
<body
  class="rmd-utils--mouse"
  style=""
>
  <div>
    <button
      aria-haspopup="menu"
      class="rmd-button rmd-button--text"
      id="main-menu"
      type="button"
    >
      Dropdown
      <i
        aria-hidden="true"
        class="rmd-icon rmd-icon--font material-icons rmd-icon-rotator rmd-icon-rotator--animate rmd-icon--after"
      >
        arrow_drop_down
      </i>
    </button>
  </div>
</body>
`;

exports[`MenuConfigurationProvider should allow for a custom header and footer for the sheet 1`] = `
<div
  aria-labelledby="main-menu"
  aria-modal="true"
  class="rmd-dialog rmd-sheet rmd-sheet--vertical rmd-sheet--raised rmd-sheet--bottom rmd-sheet--touchable-height"
  id="main-menu-menu-sheet"
  role="dialog"
  tabindex="-1"
>
  <header
    class="rmd-app-bar rmd-app-bar--normal"
  >
    <h6
      class="rmd-app-bar__title rmd-app-bar__title--no-wrap"
    >
      Custom
    </h6>
    <button
      class="rmd-button rmd-button--icon rmd-app-bar__action rmd-app-bar__action--first rmd-app-bar__action--last"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="rmd-icon rmd-icon--svg"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </button>
  </header>
  <div
    aria-labelledby="main-menu"
    class="rmd-menu"
    id="main-menu-menu"
    role="menu"
    tabindex="-1"
  >
    <ul
      class="rmd-list"
      role="none"
    >
      <li
        class="rmd-list-item rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          An Item
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        aria-haspopup="menu"
        class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
        id="sub-menu"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          Submenu
        </span>
        <i
          aria-hidden="true"
          class="rmd-icon rmd-icon--font material-icons rmd-icon-rotator rmd-icon-rotator--animate rmd-icon--after rmd-list-item__addon"
        >
          keyboard_arrow_right
        </i>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        role="none"
      >
        <a
          class="rmd-list-item rmd-list-item--clickable rmd-list-item--link rmd-menu-item"
          href="#"
          role="menuitem"
          tabindex="-1"
        >
          Link
        </a>
      </li>
      <li
        aria-disabled="true"
        class="rmd-list-item rmd-list-item--clickable rmd-list-item--disabled rmd-list-item--disabled-color rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          Disabled Item
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        class="rmd-list-item rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          Delta
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
    </ul>
  </div>
  <hr
    class="rmd-divider"
    role="separator"
  />
  <footer
    class="rmd-dialog__footer rmd-dialog__footer--flex rmd-dialog__footer--end"
  >
    <button
      class="rmd-button rmd-button--text"
      type="button"
    >
      Cancel
    </button>
  </footer>
</div>
`;

exports[`MenuConfigurationProvider should allow for a custom header and footer for the sheet 2`] = `
<div
  aria-labelledby="main-menu"
  aria-modal="true"
  class="rmd-dialog rmd-sheet rmd-sheet--vertical rmd-sheet--raised rmd-sheet--bottom rmd-sheet--touchable-height rmd-sheet--offscreen rmd-sheet--enter rmd-sheet--visible"
  id="main-menu-menu-sheet"
  role="dialog"
  tabindex="-1"
>
  <div
    aria-labelledby="main-menu"
    class="rmd-menu"
    id="main-menu-menu"
    role="menu"
    tabindex="-1"
  >
    <ul
      class="rmd-list"
      role="none"
    >
      <li
        class="rmd-list-item rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          An Item
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        aria-haspopup="menu"
        class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
        id="sub-menu"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          Submenu
        </span>
        <i
          aria-hidden="true"
          class="rmd-icon rmd-icon--font material-icons rmd-icon-rotator rmd-icon-rotator--animate rmd-icon--after rmd-list-item__addon"
        >
          keyboard_arrow_right
        </i>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        role="none"
      >
        <a
          class="rmd-list-item rmd-list-item--clickable rmd-list-item--link rmd-menu-item"
          href="#"
          role="menuitem"
          tabindex="-1"
        >
          Link
        </a>
      </li>
      <li
        aria-disabled="true"
        class="rmd-list-item rmd-list-item--clickable rmd-list-item--disabled rmd-list-item--disabled-color rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          Disabled Item
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        class="rmd-list-item rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          Delta
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
    </ul>
  </div>
</div>
`;

exports[`MenuConfigurationProvider should allow menus to be rendered in a sheet by enabling renderAsSheet 1`] = `
<body
  class="rmd-utils--mouse"
  data-rmd-noscroll=""
  style="overflow: hidden;"
>
  <div>
    <button
      aria-expanded="true"
      aria-haspopup="menu"
      class="rmd-button rmd-button--text"
      id="main-menu"
      type="button"
    >
      Dropdown
      <i
        aria-hidden="true"
        class="rmd-icon rmd-icon--font material-icons rmd-icon-rotator rmd-icon-rotator--animate rmd-icon-rotator--rotated rmd-icon--after"
      >
        arrow_drop_down
      </i>
    </button>
  </div>
  <span
    class="rmd-overlay rmd-overlay--visible rmd-overlay--clickable rmd-dialog-overlay rmd-sheet-overlay rmd-overlay--active"
    id="main-menu-menu-sheet-overlay"
    tabindex="-1"
  />
  <div
    aria-labelledby="main-menu"
    aria-modal="true"
    class="rmd-dialog rmd-sheet rmd-sheet--vertical rmd-sheet--raised rmd-sheet--bottom rmd-sheet--touchable-height rmd-sheet--offscreen rmd-sheet--enter rmd-sheet--visible"
    id="main-menu-menu-sheet"
    role="dialog"
    tabindex="-1"
  >
    <div
      aria-labelledby="main-menu"
      class="rmd-menu"
      id="main-menu-menu"
      role="menu"
      tabindex="0"
    >
      <ul
        class="rmd-list"
        role="none"
      >
        <li
          class="rmd-list-item rmd-list-item--clickable rmd-menu-item"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="rmd-list-item__text"
          >
            An Item
          </span>
          <span
            class="rmd-ripple-container"
          />
        </li>
        <li
          aria-haspopup="menu"
          class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
          id="sub-menu"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="rmd-list-item__text"
          >
            Submenu
          </span>
          <i
            aria-hidden="true"
            class="rmd-icon rmd-icon--font material-icons rmd-icon-rotator rmd-icon-rotator--animate rmd-icon--after rmd-list-item__addon"
          >
            keyboard_arrow_right
          </i>
          <span
            class="rmd-ripple-container"
          />
        </li>
        <li
          role="none"
        >
          <a
            class="rmd-list-item rmd-list-item--clickable rmd-list-item--link rmd-menu-item"
            href="#"
            role="menuitem"
            tabindex="-1"
          >
            Link
          </a>
        </li>
        <li
          aria-disabled="true"
          class="rmd-list-item rmd-list-item--clickable rmd-list-item--disabled rmd-list-item--disabled-color rmd-menu-item"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="rmd-list-item__text"
          >
            Disabled Item
          </span>
          <span
            class="rmd-ripple-container"
          />
        </li>
        <li
          class="rmd-list-item rmd-list-item--clickable rmd-menu-item"
          role="menuitem"
          tabindex="-1"
        >
          <span
            class="rmd-list-item__text"
          >
            Delta
          </span>
          <span
            class="rmd-ripple-container"
          />
        </li>
      </ul>
    </div>
  </div>
</body>
`;

exports[`MenuConfigurationProvider should allow menus to be rendered in a sheet by enabling renderAsSheet 2`] = `
<body
  class="rmd-utils--mouse"
  style=""
>
  <div>
    <button
      aria-haspopup="menu"
      class="rmd-button rmd-button--text"
      id="main-menu"
      type="button"
    >
      Dropdown
      <i
        aria-hidden="true"
        class="rmd-icon rmd-icon--font material-icons rmd-icon-rotator rmd-icon-rotator--animate rmd-icon--after"
      >
        arrow_drop_down
      </i>
    </button>
  </div>
</body>
`;

exports[`MenuConfigurationProvider should allow menus to be rendered in a sheet by enabling renderAsSheet 3`] = `
<body
  class="rmd-utils--mouse"
  style=""
>
  <div>
    <button
      aria-expanded="true"
      aria-haspopup="menu"
      class="rmd-button rmd-button--text"
      id="main-menu"
      type="button"
    >
      Dropdown
      <i
        aria-hidden="true"
        class="rmd-icon rmd-icon--font material-icons rmd-icon-rotator rmd-icon-rotator--animate rmd-icon-rotator--rotated rmd-icon--after"
      >
        arrow_drop_down
      </i>
    </button>
  </div>
  <div
    aria-labelledby="main-menu"
    class="rmd-menu rmd-menu--elevated rmd-transition--scale-y-enter rmd-transition--scale-y-enter-active"
    id="main-menu-menu"
    role="menu"
    style="left: 16px; top: 0px; position: fixed; transform-origin: 0 0;"
    tabindex="-1"
  >
    <ul
      class="rmd-list"
      role="none"
    >
      <li
        class="rmd-list-item rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          An Item
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        aria-haspopup="menu"
        class="rmd-list-item rmd-list-item--medium rmd-list-item--clickable rmd-menu-item"
        id="sub-menu"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          Submenu
        </span>
        <i
          aria-hidden="true"
          class="rmd-icon rmd-icon--font material-icons rmd-icon-rotator rmd-icon-rotator--animate rmd-icon--after rmd-list-item__addon"
        >
          keyboard_arrow_right
        </i>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        role="none"
      >
        <a
          class="rmd-list-item rmd-list-item--clickable rmd-list-item--link rmd-menu-item"
          href="#"
          role="menuitem"
          tabindex="-1"
        >
          Link
        </a>
      </li>
      <li
        aria-disabled="true"
        class="rmd-list-item rmd-list-item--clickable rmd-list-item--disabled rmd-list-item--disabled-color rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          Disabled Item
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
      <li
        class="rmd-list-item rmd-list-item--clickable rmd-menu-item"
        role="menuitem"
        tabindex="-1"
      >
        <span
          class="rmd-list-item__text"
        >
          Delta
        </span>
        <span
          class="rmd-ripple-container"
        />
      </li>
    </ul>
  </div>
</body>
`;
